athena-federation-integ-test/src/main/java/com/amazonaws/athena/connector/integ/CloudFormationTemplateProvider.java

```diff
@@ -156,6 +80,12 @@ private Map getEnvironmentVars()
      */
     protected abstract void setSpecificResource(final Stack stack);
 
+    /**
+     * Must be overridden to indicate whether the connector being tested supports a VPC configuration.
+     * @return true/false
+     */
+    protected abstract boolean isSupportedVpcConfig();
+
```

Comment:
i'm not thrilled with boolean APIs like this, can we instead have a getVPCConfig() that returns Optional? and we can use it for both? this way it avoid confusion where someone implements one correctly but not the other

Reply:
This will be removed in favor of a config file.

Reply:
I will include a public API to get the VPC config that returns an Optional.